Sources/xcresultparser/JunitXML.swift

```diff
@@ -72,56 +72,69 @@ public struct JunitXML: XmlSerializable {
             )
         }
     }
-    
+
     func createRootElement() -> XMLElement {
         let element = XMLElement(name: nodeNames.testsuitesName)
         if testReportFormat == .sonar {
             element.addAttribute(name: "version", stringValue: "1")
         }
         return element
     }
-    
+
     public var xmlString: String {
         let testsuites = createRootElement()
         let xml = XMLDocument(rootElement: testsuites)
         xml.characterEncoding = "UTF-8"
-        
+
         if testReportFormat != .sonar {
             let metrics = invocationRecord.metrics
             let testsCount = metrics.testsCount ?? 0
             testsuites.addAttribute(name: "tests", stringValue: String(testsCount))
             let testsFailedCount = metrics.testsFailedCount ?? 0
             testsuites.addAttribute(name: "failures", stringValue: String(testsFailedCount))
         }
-        
-        let testAction = invocationRecord.actions.first { $0.schemeCommandName == "Test" }
-        guard let testsId = testAction?.actionResult.testsRef?.id,
-              let testPlanRun = resultFile.getTestPlanRunSummaries(id: testsId) else {
-                  return xml.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
-              }
-        
-        if testReportFormat != .sonar {
-            if let date = testAction?.startedTime, let ended = testAction?.endedTime {
-                let duration = ended.timeIntervalSince(date)
-                testsuites.addAttribute(name: "time", stringValue: numFormatter.unwrappedString(for: duration))   
-            }
+
+        let testActions = invocationRecord.actions.filter({ $0.schemeCommandName == "Test" })
+        guard !testActions.isEmpty else {
+            return xml.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
         }
-        let testPlanRunSummaries = testPlanRun.summaries
-        let failureSummaries = invocationRecord.issues.testFailureSummaries
-        
-        for thisSummary in testPlanRunSummaries {
-            for thisTestableSummary in thisSummary.testableSummaries {
-                for group in thisTestableSummary.tests {
-                    for testsuite in createTestSuite(group, failureSummaries: failureSummaries) {
-                        testsuites.addChild(testsuite)
+
+        var overallTestSuiteDuration = 0.0
+        for testAction in testActions {
+            guard let testsId = testAction.actionResult.testsRef?.id,
+                  let testPlanRun = resultFile.getTestPlanRunSummaries(id: testsId) else {
+                      return xml.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
```

Comment:
Shouldn't we rather "continue" here.
Maybe it's only this loop iteration and the next loop iteration is fine?

Reply:
Oh yeah definitely, I'll get it updated

Reply:
Updated